orc8r/gateway/python/magma/state/state_replicator.py

```diff
@@ -135,11 +135,17 @@ async def _collect_states_to_replicate(self):
                     continue
 
                 redis_state = redis_dict.get(key)
-                if redis_dict.state_format == PROTO_FORMAT:
-                    state_to_serialize = MessageToDict(redis_state)
-                    serialized_json_state = json.dumps(state_to_serialize)
-                else:
-                    serialized_json_state = jsonpickle.encode(redis_state)
+                try:
+                    if redis_dict.state_format == PROTO_FORMAT:
+                        state_to_serialize = MessageToDict(redis_state)
+                        serialized_json_state = json.dumps(state_to_serialize)
+                    else:
+                        serialized_json_state = jsonpickle.encode(redis_state)
+                except Exception as e:  # pylint: disable=broad-except
+                    logging.error("Found bad state for %s for %s, not "
```

Comment:
IMO, This will result in silent failure. We should probably do more than this. Either zeroing out the values or dealing with this from application perspective. known failure state is better than unknown successful state. This should be either seen as a replication failure event which lets user make the determination on what is the bad state and how to go about resolving it.